### DISCUSSION TITLE
Is HLS able to print out the version of a plugin that it's using?

### DISCUSSION THREAD

**Discussion by jonathanknowles** [haskell/haskell-language-server#3741] [Q&A] [answered]
Specifically, I'd like to know the version of **`fourmolu`** that HLS is using.

I'm using HLS version `2.0.0.1`.

**Problem**: when I attempt to reformat a file with HLS and `fourmolu`, it seems to be using a much older version of `fourmolu` than the one available in my PATH (which is `0.13.1.0`).

I suspect this is the case, because the HLS version of `fourmolu` ignores various configuration options (in `fourmolu.yaml`) that were only introduced in more recent versions of `fourmolu`. But when I reformat with `fourmolu` on the command line (with the `fourmolu` on my PATH), these configuration options are respected.

**Comment by fendor** [comment]
Unfortunately, there is no easy way to do that, currently. Currently the "easiest" way is to check out the sources of the HLS release `2.0.0.1` and run `cabal build --dry-run` or `cabal freeze` and then look up the version in `dist-newstyle/cache/plan.json` or `cabal.project.freeze`. 

See https://github.com/haskell/haskell-language-server/issues/3660 which describes a potential solution and implementation roadmap.

**Reply by jonathanknowles** [reply]
Hi @fendor 

Many thanks for your very swift reply!

**Comment by michaelpj** [comment, marked as answer]
This is now shown when you run `haskell-language-server --list-plugins`

**Reply by jonathanknowles** [reply]
Many thanks @michaelpj!

I've just tested this with `haskell-language-server` version `2.6.0.0` and it seems to work nicely. For example:
```bash
$ haskell-language-server --list-plugins | grep fourmolu       
fourmolu:
    Provides formatting of Haskell files via fourmolu. Built with fourmolu-0.14.0.0
```
Many thanks for the update!
